### PR TITLE
Updates balance in tip banner

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/7047

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
covered with tests

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by NejcZdovc** [comment]
CI failed on `test-install`. Not related

File: browser/ui/webui/brave_tip_ui.cc [hunk 1]
```diff
@@ -477,3 +484,22 @@ void RewardsTipDOMHandler::OnlyAnonWallet(const base::ListValue* args) {
       "brave_rewards_tip.onlyAnonWallet",
       base::Value(allow));
 }
+
+void RewardsTipDOMHandler::OnReconcileComplete(
+    brave_rewards::RewardsService* rewards_service,
+    unsigned int result,
```
Comment (masparrow) [inline browser/ui/webui/brave_tip_ui.cc]: const uint32_t - or would that be too far reaching now?

File: browser/ui/webui/brave_tip_ui.cc [hunk 2]
```diff
@@ -477,3 +484,22 @@ void RewardsTipDOMHandler::OnlyAnonWallet(const base::ListValue* args) {
...
+    const std::string& viewing_id,
+    const std::string& probi,
+    const int32_t type) {
+  if (!web_ui()->CanCallJavascript()) {
+     return;
+  }
+
+  base::DictionaryValue complete;
+  complete.SetKey("result", base::Value(static_cast<int>(result)));
```
Comment (masparrow) [inline browser/ui/webui/brave_tip_ui.cc]: `<int>` should be `<int32_t> or the right 'int' storage type expected in the dictionary

File: browser/ui/webui/brave_tip_ui.cc (hunk 1 above)
Comment (NejcZdovc) [reply browser/ui/webui/brave_tip_ui.cc]: this is an existing observer, so we would need to change all observes to have this change

File: browser/ui/webui/brave_tip_ui.cc (hunk 2 above)
Comment (NejcZdovc) [reply browser/ui/webui/brave_tip_ui.cc]: ` base::Value` accepts int by default that's why I casted it to int directly 
https://cs.chromium.org/chromium/src/base/values.h?type=cs&g=0&l=130

**Pull request merged**